cool-core/src/main/java/com/nus/cool/core/io/writestore/MetaHashFieldWS.java

```diff
@@ -184,15 +186,41 @@ public int writeTo(DataOutput out) throws IOException {
   }
 
   @Override
-  public String toString() {
-    return "HashMetaField: " + hashToTerm.entrySet().stream().map(x -> x.getKey() + "-" + x.getValue()).collect(Collectors.toList());
+  public int writeCubeMeta(DataOutput out) throws IOException {
+    int bytesWritten = 0;
+    if (this.fieldType == FieldType.Segment
+      || this.fieldType == FieldType.Action
+      || this.fieldType == FieldType.UserKey) {
+        try (DataOutputBuffer buffer = new DataOutputBuffer()) {
+          buffer.writeInt(this.fieldValues.size());
+          int off = 0;
+  
+          for (String s : this.fieldValues) {
```

Comment:
The latest version of dev stores the hashToTerm here. Please check the order of the storage